### PR TITLE
Colin

### PR DESCRIPTION
**Pull request by coldog86**
Unit tests and integration test for entry are done. Entry with a season ticket is not part of this.

### CODE REVIEW

File: src/AdhocTicketTests [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+            
+    @Before
+	public void setUp() {
+            mockCarpark = mock(Carpark.class);
+            mockGate = mock(IGate.class);
+            mockIS = mock(CarSensor.class);
+            mockOS = mock(CarSensor.class);
+            mockUI = mock(IEntryUI.class);
+            mockTicket = mock(IAdhocTicket.class);
+            mockEntry = new EntryController(mockCarpark, mockGate, mockOS, mockIS, mockUI);
```
Comment (Abrody01) [inline src/AdhocTicketTests]: Recommended to change variable name from mockEntry to testEntry, This way it is clearer what is being mocked vs tested.

File: src/AdhocTicketTests [hunk 2]
```diff
@@ -0,0 +1,106 @@
+import bcccp.carpark.CarSensor;
```
Comment (Abrody01) [inline src/AdhocTicketTests:1]: I don't think you need to have this test at all.

File: src/EntryControllerTest [hunk 3]
```diff
@@ -0,0 +1,337 @@
...
+@FixMethodOrder(MethodSorters.NAME_ASCENDING)
+public class EntryControllerTest {
+    
+            Carpark mockCarpark;
+            IGate mockGate;
+            CarSensor mockIS;
+            CarSensor mockOS;
+            IEntryUI mockUI;
+            IAdhocTicket mockTicket;
+            EntryController mockEntry;
```
Comment (Abrody01) [inline src/EntryControllerTest]: Recommend changing "mockEntry" to "testEntry" so it is more clear what you are testing

File: src/EntryControllerTest [hunk 4]
```diff
@@ -0,0 +1,337 @@
...
+        
+        when(mockOS.carIsDetected()).thenReturn(true);
+        when(mockOS.getId()).thenReturn("Entry Outside Sensor");
+        mockEntry.carEventDetected("Entry Outside Sensor", true);
+        
+        assertEquals(mockEntry.getSTATE().toString(), "ENTERING");//verify transition state
+        
+    }
+    
+
```
Comment (Abrody01) [inline src/EntryControllerTest]: I would recommend splitting this file out to a few components, one for 'state transition' and one for the constructor.

**Review by Abrody01 — CHANGES_REQUESTED** [review]
I think you should also the user entering a season ticket along with the button pushed test's. This is in the use case as part 3.2 and 7.2

File: src/EntryControllerTest (hunk 3 above)
Comment (coldog86) [reply src/EntryControllerTest]: resolved

File: src/EntryControllerTest (hunk 4 above)
Comment (coldog86) [reply src/EntryControllerTest]: resolved

**Review by Abrody01 — APPROVED** [review]
All good to go now great work Colin

**Pull request merged**